protocol/standard-l2-genesis.md

```diff
@@ -153,8 +185,33 @@ The additional deposit gas is the only additional resource usage and its covered
 at the bottom of this document.
 
 This approach expands the ABI of the `L1Block` contract, meaning that automatically generated solidity dispatcher
-will binary search over the possible function selectors, consuming a bit more gas. This is something that we could
-avoid by writing a custom dispatcher in a fallback function or by grinding for names of functions.
+will binary search over the possible function selectors, consuming a bit more gas.
+
+## Implications for the Predeploy Releases Process
+
+### L2Genesis Generation
+
+When a new predeploy release is created, the bytcode from each predeploy should be placed into a
+an new autogenerated library which resembles the following:
+
+```solidity
+library HolocenePredeploys {
+  bytes constant L1Block = hex"...";
+  ...
+}
+```
+
+The `L2Genesis.s.sol` solidity script, will have additional functionality so that it can
+optionally generate the L2 state from the current commit as it currently does using
+`vm.getDeployedCode()`, or retrieve the code from the specified library.
+
+### Upgrade Process
+
+Note that this design modifies how L2 upgrade auth is managed (moving that management into a single
+storage slot on L1), but does not change how upgrades to predeploy contracts are performed.
+Predeploy upgrades can still be done either via a `TransactionDeposited()` event, or a [network
+upgrade automation
+transaction](https://github.com/ethereum-optimism/specs/blob/9f7226be064be0c87f90cbc6be6b0a4b4f58656a/specs/protocol/derivation.md#network-upgrade-automation-transactions).
```

Comment:
We should have a preference here. My understanding is the preference is default to network upgrade automation
transactions, and only use deposit transactions for incident response?

Reply:
TODO: Elaborate on this. Include recco about using fresh deposit accounts to get same impl addresses on all chains.

Reply:
0cf08e8f7b0d